src/js/utils/JobUtil.js

```diff
@@ -72,16 +72,22 @@ const JobUtil = {
       cmd: general.cmd,
       cpus: general.cpus,
       mem: general.mem,
-      disk: general.disk
+      disk: general.disk,
+      docker
     });
 
-    if (docker && docker.image) {
-      Object.assign(spec.run, { docker });
-      if (docker_parameters != null && docker_parameters.items != null) {
-        spec.run.docker.parameters = docker_parameters.items;
-      } else {
-        spec.run.docker.parameters = [];
-      }
+    if (docker_parameters != null && docker_parameters.items != null) {
+      spec.run.docker.parameters = docker_parameters.items;
+    } else {
+      spec.run.docker.parameters = [];
+    }
+
+    if (
+      spec.run.docker &&
+      !spec.run.docker.image &&
+      spec.run.docker.parameters.length === 0
+    ) {
+      delete spec.run.docker;
```

Comment:
Why do we want to delete it?

Reply:
If we don't delete it if the user added parameters there is no way they are able to produce a valid spec without touching the JSON.